libsolidity/analysis/StaticAnalyzer.cpp

```diff
@@ -156,6 +156,22 @@ bool StaticAnalyzer::visit(VariableDeclaration const& _variable)
 			// This is not a no-op, the entry might pre-exist.
 			m_localVarUseCount[make_pair(_variable.id(), &_variable)] += 0;
 	}
+
+	// This excludes struct members on purpose.
+	if (_variable.isStateVariable() || _variable.referenceLocation() == VariableDeclaration::Location::Storage)
+	{
+		TypePointer varType = _variable.annotation().type;
+		for (Type const* subtype: varType->allSubtypes())
+			if (subtype->storageSizeUpperBound() >= (bigint(1) << 64))
+			{
+				string message = "Type " + util::escapeAndQuoteString(subtype->toString(true)) +
```

Comment:
The side-effect of setting `_short` to `true` in `Type::toString(true)` is that now error messages contain stuff like `"struct C.P[104]"` where there used to be just `"C.P[104]"`. Which is not great since it makes it look like the whole thing is a struct (and there's no parenthesis to make it unambiguous).

But maybe we should consider it a bug in `toString()`?

Reply:
Yes, looks confusing. I used `Type::toString(true)`, because we do so in other places (and because `canonicalName()` was crashing).

Maybe we should open a separate ticket indeed.

Reply:
Actually, after looking at #9791 I think that maybe it's fine as is. Array of structs looks like the only case where it looks a little ambiguous due to the space.

On the other hand, it's probably useful to know if it's a struct, enum, contract or something else.